lib/index.js

```diff
@@ -96,16 +96,38 @@ function Socket (type, opts) {
             break;
     }
 
-    this.binding = nn.Socket(this.af_domain, this.protocol);
-    this.queue = [];
+  this.binding = nn.Socket(this.af_domain, this.protocol);
+  this.queue = [];
 
-    for(var sokopt in sol){
-        if(opts.hasOwnProperty(sokopt)) this[sokopt](opts[sokopt]);
+  /* subscription filters */
+  this.channels = {};
+
+  /* subscription handling at initialization */
+  if (opts.hasOwnProperty('chan')) {
+    if (typeof opts.chan == 'string') {
+      opts.chan.split(',').forEach(subscriber.bind(this))
+    } else {
+      opts.chan.forEach(subscriber.bind(this))
     }
```

Comment:
In both cases of the conditional, the code would be more robust if we first called `.trim()` on what was being passed to subscriber.  In the case of splitting, we should filter out empty elements, after trimming.  If you map trim across all elements, then filter on the first branch, that would be nice.